lib/chef/resource/template.rb

```diff
@@ -60,29 +57,9 @@ def source(file = nil)
         )
       end
 
-      def variables(args = nil)
-        set_or_return(
-          :variables,
-          args,
-          :kind_of => [ Hash ]
-        )
-      end
-
-      def cookbook(args = nil)
-        set_or_return(
-          :cookbook,
-          args,
-          :kind_of => [ String ]
-        )
-      end
-
-      def local(args = nil)
-        set_or_return(
-          :local,
-          args,
-          :kind_of => [ TrueClass, FalseClass ]
-        )
-      end
+      property :variables, Hash, default: {}
```

Comment:
this is a frozen hash default.

```ruby
template "whatever.erb" do
   variables[:aliases] ||= {}
end
```

That mutates the Hash which is the default value.

You need to get a new Hash object for every new template resource.

Frozen default values probably just caught a huge bug here.

Reply:
`default: lazy { Hash.new }` maybe?

Reply:
yeah validated that produces different objects.

i'd almost go so far as to say that `default: {}` is always a bug and should be replaced with this (and any default hash, even with default keys should be lazied this way).